core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2555,6 +2555,15 @@ private[spark] object Utils extends Logging {
       sparkJars.map(_.split(",")).map(_.filter(_.nonEmpty)).toSeq.flatten
     }
   }
+
+  private[util] val REDACTION_REPLACEMENT_TEXT = "*********(redacted)"
+  def redact(conf: SparkConf)(kv: (String, String)): (String, String) = {
+    val redactionPattern = conf.get(SECRET_REDACTION_PATTERN).r
```

Comment:
What part do you think is expensive? Going through all the configuration properties and matching them with the regex?
If so, I agree. However, that has to be done somewhere. All the callers of this function have a `SparkConf` that they want stuff redacted from. So, if this function accepts a list of tuples, they have to run the regex check to find that list first before sending it to `redact()`. So, overall, unless I am missing something, I don't think we can avoid the expense.